test/test.c

```diff
@@ -391,57 +391,56 @@ bool SetAttributes(HCOSE hHandle, const cn_cbor * pAttributes, int which, int ms
 		switch (msgType) {
 #if INCLUDE_MAC
 		case Attributes_MAC_protected:
-			f = COSE_Mac_map_put_int((HCOSE_MAC)hHandle, keyNew, pValueNew, which, NULL);
+			fRet = COSE_Mac_map_put_int((HCOSE_MAC)hHandle, keyNew, pValueNew, which, NULL);
 			break;
```

Comment:
This should be "fRet &= ..."  as well as the rest
Initialize "fRet = true"

If there re no items set, then fRet is never changed from the intial false value.